src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -197,34 +197,43 @@ public void Start()
                 }
 
                 // STEP 2: Open the socket connection to the remote host
-                var cts = new CancellationTokenSource();
-                var tok = cts.Token;
-                
-                tok.Register(() =>
-                {
-                    _client.Close();
-                });
+                var cts = new CancellationTokenSource(ConnectTimeout);
+                cts.CancelAfter(ConnectTimeout);
 
+                using (cts.Token.Register(() => { _client.Close(); }))
+                {
+                    _client.Dispose();
```

Comment:
This is not good because it will dispose the client before it is used below (Maybe you meant to put it inside the callback?)